26.1.0-rc2: error: implicit declaration of function 'strerror_l' is invalid in C99

clang-10.0.1 fails to compile 26.1.0-rc2 on FreeBSD 12.2:

```
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:129:4: error: implicit declaration of function 'strerror_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                        berr(LOG_ERROR, "exiting due to poll error");
                        ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:129:4: note: did you mean 'strerror_r'?
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/include/string.h:91:6: note: 'strerror_r' declared here
int      strerror_r(int, char *, size_t);
         ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:129:4: warning: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
                        berr(LOG_ERROR, "exiting due to poll error");
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:15: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                            ^        ~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:145:5: error: implicit declaration of function 'strerror_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                                berr(LOG_ERROR,
                                ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:145:5: warning: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
                                berr(LOG_ERROR,
                                ^~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:15: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                            ^        ~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:276:3: error: implicit declaration of function 'strerror_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                berr(LOG_ERROR, "socketpair");
                ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:276:3: warning: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
                berr(LOG_ERROR, "socketpair");
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:15: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                            ^        ~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:288:3: error: implicit declaration of function 'strerror_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                berr(LOG_ERROR, "could not open %s sndio device", devname);
                ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:288:3: warning: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
                berr(LOG_ERROR, "could not open %s sndio device", devname);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:15: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                            ^        ~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:301:3: error: implicit declaration of function 'strerror_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                berr(LOG_ERROR, "could not set parameters for %s sndio device",
                ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:301:3: warning: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
                berr(LOG_ERROR, "could not set parameters for %s sndio device",
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:15: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                            ^        ~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:309:3: error: implicit declaration of function 'strerror_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                berr(LOG_ERROR, "could not start recording on %s sndio device",
                ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:309:3: warning: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
                berr(LOG_ERROR, "could not start recording on %s sndio device",
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:15: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                            ^        ~~~~~~~~~~~~~~~~~~~~~~~
7 warnings and 6 errors generated.
```

<!-- READ THIS FIRST -->
<!-- The OBS Studio GitHub issue tracker is **ONLY** to be -->
<!-- used for reporting Bugs that have replication steps. -->

<!-- You can post Feature Requests here: https://ideas.obsproject.com/ -->
<!-- Get help for Support Issues here: https://obsproject.com/help -->

<!--- Provide a general summary of the issue in the Title above -->

## Platform

<!-- If you are on Linux and installed using a package, please list the package type. -->
Operating system and version: FreeBSD 12.2
OBS Studio version: 26.1.0-rc2

## Expected Behavior
compilation succeeds

## Current Behavior
<!--- Tell us what happens instead of the expected behavior. -->
<!--- Please include a log file here if possible. -->
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:129:4: error: implicit declaration of function 'strerror_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                        berr(LOG_ERROR, "exiting due to poll error");
                        ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:129:4: note: did you mean 'strerror_r'?
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/include/string.h:91:6: note: 'strerror_r' declared here
int      strerror_r(int, char *, size_t);
         ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:129:4: warning: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
                        berr(LOG_ERROR, "exiting due to poll error");
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:15: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                            ^        ~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:145:5: error: implicit declaration of function 'strerror_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                                berr(LOG_ERROR,
                                ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:145:5: warning: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
                                berr(LOG_ERROR,
                                ^~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:15: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                            ^        ~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:276:3: error: implicit declaration of function 'strerror_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                berr(LOG_ERROR, "socketpair");
                ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:276:3: warning: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
                berr(LOG_ERROR, "socketpair");
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:15: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                            ^        ~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:288:3: error: implicit declaration of function 'strerror_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                berr(LOG_ERROR, "could not open %s sndio device", devname);
                ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:288:3: warning: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
                berr(LOG_ERROR, "could not open %s sndio device", devname);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:15: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                            ^        ~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:301:3: error: implicit declaration of function 'strerror_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                berr(LOG_ERROR, "could not set parameters for %s sndio device",
                ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:301:3: warning: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
                berr(LOG_ERROR, "could not set parameters for %s sndio device",
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:15: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                            ^        ~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:309:3: error: implicit declaration of function 'strerror_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                berr(LOG_ERROR, "could not start recording on %s sndio device",
                ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:24: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                                     ^
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:309:3: warning: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
                berr(LOG_ERROR, "could not start recording on %s sndio device",
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/multimedia/obs-studio/work/obs-studio-26.1.0-rc2/plugins/sndio/sndio-input.c:41:15: note: expanded from macro 'berr'
                const char *errstr = strerror_l(errno, NULL);   \
                            ^        ~~~~~~~~~~~~~~~~~~~~~~~
7 warnings and 6 errors generated.

## Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant. -->
<!--- Screenshots and video are encouraged if applicable. -->
1. cmake .
2. gmake
3.
4.

## Additional information
<!--- Not obligatory, but provide any additional details or information -->
<!--- that you feel might be relevant to the issue -->

--- a/plugins/sndio/sndio-input.c
+++ b/plugins/sndio/sndio-input.c
@@ -38,7 +38,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #define berr(level, msg, ...)                                   \
 	do {                                                    \
-		const char *errstr = strerror_l(errno, NULL);   \
+		const char *errstr = strerror(errno);           \
 		blog(level, msg ": %s", ##__VA_ARGS__, errstr); \
 	} while (0)
 

